packages/gatsby-image/src/index.js

```diff
@@ -703,6 +718,16 @@ Image.defaultProps = {
   loading: `lazy`,
 }
 
+const imageMediaTypes = {
+  apng: `image/apng`,
+  flif: `image/flif`,
+  gif: `image/gif`,
+  jpg: `image/jpg`,
+  jpeg: `image/jpg`,
+  png: `image/png`,
+  webp: `image/webp`,
+}
```

Comment:
Where are you sourcing this list from? Perhaps you should mirror [what MDN lists](https://developer.mozilla.org/en-US/docs/Web/Media/Formats/Image_types) (probably fine to omit BMP and ICO?), other uncommon extensions may not be relevant to include/support, I'm not fussed.

`flif` shouldn't be there. It [doesn't have any actual browser support does it?](https://caniuse.com/#search=flif) Only polyfill, which I can understand you wanting to allow support for, but that won't help AFAIK, [you're providing a type hint to the browser for source selection, it may even be ignored as a result](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/source) (I haven't looked at the polyfill, but I assume it's not modifying the browsers native user agent?):

> If the `type` attribute isn't specified, the media's type is retrieved from the server and checked to see if the user agent can handle it; if it can't be rendered, the next `<source>` is checked. If the `type` attribute is specified, it's compared against the types the user agent can present, and if it's not recognized, the server doesn't even get queried; instead, the next `<source>` element is checked at once.

Feel free to verify what happens and report back on that.

More valid types that have rather limited browser support are [JPEG 2000](https://caniuse.com/#search=jpeg%202000), [JPEG XR](https://caniuse.com/#search=jpeg%20xr) and [AVIF](https://caniuse.com/#search=avif). Each has their own mime types.

Finally, do keep in mind that as `webp` type is presented first, it will be used if supported by the browser in preference of the other / original source. `gatsby-image` is generally aimed at serving `webp` as the specialized/niche format, with a broadly supported type as fallback. I wouldn't see any sane reason to have `flif` here due to that.